unlock-app/src/components/lock/Lock.js

```diff
@@ -0,0 +1,133 @@
+import UnlockPropTypes from '../../propTypes'
+import uniqid from 'uniqid'
+import PropTypes from 'prop-types'
+import styled from 'styled-components'
+import React from 'react'
+import Balance from '../helpers/Balance'
+import { connect } from 'react-redux'
+
+import { purchaseKey } from '../../actions/key'
+
+export const Lock = ({ lock, account, keys, purchaseKey }) => {
+  if(!account) {
+    return null
+  }
+
+  let accessKey = Object.values(keys).find((key) => (
+    key.lockAddress === lock.address && key.owner === account.address
+  ))
+  if (!accessKey) {
+    accessKey = {
+      id: uniqid(),
+      lockAddress: lock.address,
+      owner: account.address,
+    }
+  }
+
+  let purchaseButton = <PurchaseButton onClick={() => { purchaseKey(accessKey) }}>Purchase</PurchaseButton>
+
+  if (!accessKey.transaction) {
+    // No transaction attached to the key. What is it?
+    // Maybe we just lost track of that transaction?
+    // Is the key valid?
+  } else if (accessKey.transaction.status === 'mined') {
+    purchaseButton = <PurchaseButton>Mined! Confirming... {accessKey.transaction.confirmations}</PurchaseButton>
+    // Key transaction was mined: it is mined, let's look at confirmations
+  } else if (accessKey.transaction.status === 'submitted') {
+    purchaseButton = <PurchaseButton>Submitted!</PurchaseButton>
+    // Ok, let's wait and not show another button!
+  }
+
+  return (
+    <Wrapper>
+      <Name>{lock.name}</Name>
+      <EtherPrice><Balance amount={lock.keyPrice} /></EtherPrice>
+      {lock.fiatPrice &&
+      <FiatPrice>${lock.fiatPrice}</FiatPrice>
```

Comment:
Should this be indented?

Reply:
I thought so but my eslint config keeps unindented it. Not sure why. I still need to update the eslint config...